File: 1028.cpp
#include <iostream>
#include <iomanip>

using namespace std;

int main(void)
{
    int n, m, q, i;
    int anterior, atual, resto;

    cin >> q;
    for (i = 0; i < q; ++i) {
        cin >> n >> m;
        anterior = n;
        atual = m;
        resto = anterior % atual;
        while (resto != 0) {
            anterior = atual;
            atual = resto;
            resto = anterior % atual;
        }
        cout << atual << endl;
    }
    return 0;
}

File: 5-matematica/1028.py
n = int(input())
for i in range(n):
    e = str(input()).split()
    n = int(e[0])
    m = int(e[1])
    anterior = n
    atual = m
    resto = anterior % atual
    while resto != 0:
        anterior = atual
        atual = resto
        resto = anterior % atual
    print(atual)
